handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/GetAccount.scala

```diff
@@ -0,0 +1,112 @@
+    zuoraGet.get[PaymentMethodResponse](uri"object/payment-method/$paymentMethodId")
+
+trait GetAccount:
+  def get(subscriptionNumber: String): ZIO[GetAccount, String, GetAccountResponse]
+
+  def getPaymentMethod(paymentMethodId: String): ZIO[GetAccount, String, PaymentMethodResponse]
+
+object GetAccount {
+  case class PaymentMethodResponse(NumConsecutiveFailures: Int)
+
+  given JsonDecoder[PaymentMethodResponse] = DeriveJsonDecoder.gen[PaymentMethodResponse]
+
+  case class GetAccountResponse(
+    basicInfo: BasicInfo,
+    subscriptions: List[AccountSubscription],
+  )
+
+  case class AccountSubscription(
+    id: String
+  )
+
+  case class BasicInfo(
+    defaultPaymentMethod: WireDefaultPaymentMethod,
+    balance: BigDecimal,
+    currency: String
+  )
+
+  /* The zuora API has slightly different responses for ratePlans in the catalogue and when querying a subscription, e.g.: productRatePlanId in the subscription response and id in the catalogue response */
+
+  case class ZuoraSubscription(
+    ratePlans: List[ZuoraRatePlan]
+  )
+
+  case class ZuoraRatePlan(
+    productRatePlanId: String // TBC what field should we check in the rateplan to know it's a recurring contribution?
+    /*
+    {
+                    "productId": "2c92c0f84b786da2014b91d3629b4298",//DEV
+                    "productName": "Digital Pack",
+                    "productSku": "SKU-00000022",
+                    "productRatePlanId": "2c92c0f94bbffaaa014bc6a4212e205b",//DEV
+                    "ratePlanName": "Digital Pack Annual"
+                }*/
+  )
+
+  given JsonDecoder[GetAccountResponse] = DeriveJsonDecoder.gen
+  given JsonDecoder[AccountSubscription] = DeriveJsonDecoder.gen
+  given JsonDecoder[ZuoraSubscription] = DeriveJsonDecoder.gen
+  given JsonDecoder[ZuoraRatePlan] = DeriveJsonDecoder.gen
+  given JsonDecoder[BasicInfo] = DeriveJsonDecoder.gen
+
+  def get(accountNumber: String): ZIO[GetAccount, String, GetAccountResponse] =
+    ZIO.serviceWithZIO[GetAccount](_.get(accountNumber))
+
+  def getPaymentMethod(paymentMethodId: String): ZIO[GetAccount, String, PaymentMethodResponse] = ZIO.serviceWithZIO[GetAccount](_.getPaymentMethod(paymentMethodId))
+}
+
+case class DefaultPaymentMethod(
+  id: String,
+  creditCardExpirationDate: LocalDate,
+)
+
+case class WireDefaultPaymentMethod(
+  id: String,
+  creditCardExpirationMonth: Option[Int],
+  creditCardExpirationYear: Option[Int],
+)
+
+given JsonDecoder[WireDefaultPaymentMethod] = DeriveJsonDecoder.gen[WireDefaultPaymentMethod]
+
+/*
+* Don't use discount percentage from product catalogue,
+* because it can be overridden so the default value is unreliable.
+*/
+// price can be null, this is only for percentage discounts, so default to 0 rather than option handling
+case class ZuoraPricing(currency: String, price: BigDecimal = 0.000000000)
+
+object ZuoraPricing {
+  given JsonDecoder[ZuoraPricing] = DeriveJsonDecoder.gen[ZuoraPricing]
+}
```

Comment:
I realise you've handled it in the business logic later, but on balance I would prefer it to be a deserialisation concern to keep the business logic more focussed.  It does require writing custom decoder though which can be a pain.  What do you think of something like this?
```suggestion
case class ZuoraPricing(currency: String, priceMinorUnits: Int)

object ZuoraPricing {

  private case class ZuoraPricingWire(currency: String, price: BigDecimal = 0.000000000)
  
  given JsonDecoder[ZuoraPricing] = DeriveJsonDecoder.gen[ZuoraPricingWire].map {
    case ZuoraPricingWire(currency, price) => ZuoraPricing(currency, (price.toDouble * 100).toInt)
  }
}
```

Reply:
Good suggestion, have amended